src/Components/DeepZoom/DeepZoom.tsx

```diff
@@ -135,10 +131,19 @@ const DeepZoom: React.FC<React.PropsWithChildren<DeepZoomProps>> = ({
 
   const isMounted = useDidMount()
 
+  useEffect(() => {
+    return () => {
+      if (osdViewerRef.current && osdViewerRef.current.destroy) {
+        osdViewerRef.current.destroy()
+        osdViewerRef.current = null
+      }
+    }
```

Comment:
I just learned about the cleanup function in `useEffect` hooks last week, so it's cool to see one in the wild 🎉

Reply:
First time ever seeing it

Reply:
are we doing this elsewhere @iskounen or did you read about it in the docs?